src/toil/test/src/jobTest.py

```diff
@@ -254,6 +254,112 @@ def checkFollowOnEdgeCycleDetection(fNode, tNode):
                 and (fNode, tNode) not in childEdges and (fNode, tNode) not in followOnEdges):
                 checkFollowOnEdgeCycleDetection(fNode, tNode)
 
+    def testNewCheckpointIsLeafVertexNonRootCase(self):
+        """
+        Test for issue #1465: Detection of checkpoint jobs that are not leaf vertices
+        identifies leaf vertices incorrectly
+
+        Test verification of new checkpoint jobs being leaf verticies,
+        starting with the following baseline workflow:
+
+            Parent
+              |
+            Child # Checkpoint=True
+
+        """
+
+        def createWorkflow():
+            rootJob = Job.wrapJobFn(simpleJobFn, "Parent")
+            childCheckpointJob = rootJob.addChildJobFn(simpleJobFn, "Child", checkpoint=True)
+            return rootJob, childCheckpointJob
+
+        self.runNewCheckpointIsLeafVertexTest(createWorkflow)
+
+    def testNewCheckpointIsLeafVertexRootCase(self):
+        """
+        Test for issue #1466: Detection of checkpoint jobs that are not leaf vertices
+                              omits the workflow root job
+
+        Test verification of a new checkpoint job being leaf vertex,
+        starting with a baseline workflow of a single, root job:
+
+            Root # Checkpoint=True
+
+        """
+
+        def createWorkflow():
+            rootJob = Job.wrapJobFn(simpleJobFn, "Root", checkpoint=True)
+            return rootJob, rootJob
+
+        self.runNewCheckpointIsLeafVertexTest(createWorkflow)
+
+    def runNewCheckpointIsLeafVertexTest(self, createWorkflowFn):
+        """
+        Test verification that a checkpoint job is a leaf vertex using both
+        valid and invalid cases.
+
+        :param createWorkflowFn: function to create and new workflow and return a tuple of:
+
+                                 0) the workflow root job
+                                 1) a checkpoint job to test within the workflow
+
+        """
+
+        print('Test checkpoint job that is a leaf vertex')
```

Comment:
Hi CJ,

Thanks for the helpful comment.
Yes, logging is the preferred way to output informational text.
As background, when adding these tests originally, I checked whether the test (jobTest.py) used logging or print for informational/diagnostic messages. I found no use of logging and a few print statements. I then incorrectly inferred that this is the CGL/Toil convention. Yet, based on viewing other tests since then, I see that logging is clearly the convention, as it should be.
I added the commonly used test logging setup in JobTest.setUpClass, then changed the print statements I had added to this test to logging statements instead.
I did not change the other few, preexisting print statements in jobTest.py as they are not associated with this PR. I would be happy to update these preexisting uses of print as well, yet that is probably best not done as part of this in this PR, IMHO.